File: abac/swig/python/abac_prover.py
#!/usr/bin/python

import getopt
import sys
from ABAC import *
import pprint

def usage():
    print "Usage: abac_prover.py \\"
    print "        --keystore <keystore> \\"
    print "        --role <role> --principal"
    print "    loads the keystore and runs the query role <-?- principal"
    sys.exit(1)

pp = pprint.PrettyPrinter(indent=4)

keystore = ''
role = ''
principal = ''

try:
    opts, args = getopt.getopt(sys.argv[1:], '', ['keystore=', 'role=', 'principal='])
except getopt.GetoptError, err:
    print str(err)
    sys.exit(1)

for o, a in opts:
    if o == '--keystore':
        keystore = a
    elif o == '--role':
        role = a
    elif o == '--principal':
        principal = a
    else:
        assert False, "WAT"

if keystore == '' or role == '' or principal == '':
    usage()

# code begins here! sorry about that

ctx = Context()
ctx.load_directory(keystore)

(success, credentials) = ctx.query(role, principal)

if success:
    print "success"
else:
    print "fail, here's a partial proof"

for credential in credentials:
    print "credential %s <- %s" % (credential.head().string(), credential.tail().string())
    # pp.pprint(credential)

File: abac/swig/perl/abac_prover.pl
#!/usr/bin/perl

use strict;
use ABAC;
use Getopt::Long;
use Data::Dumper;
$Data::Dumper::Indent = 1;

my ($keystore, $role, $principal);
GetOptions(
    'keystore=s'    => \$keystore,
    'role=s'        => \$role,
    'principal=s'   => \$principal,
) || usage();

usage() unless defined $keystore && defined $role && defined $principal;

# code starts here

my $ctx = ABAC::Context->new;
$ctx->load_directory($keystore);

my ($success, $credentials) = $ctx->query($role, $principal);

if ($success) {
    print "Success\n";
}
else {
    print "Fail, here's a partial proof\n";
}

foreach my $credential (@$credentials) {
    printf "credential %s <- %s\n",
        $credential->head->string,
        $credential->tail->string;
}

sub usage {
    print "Usage: $0 \\\n";
    print "        --keystore <keystore> \\\n";
    print "        --role <role> --principal <principal>\n";
    print "    loads the keystore and runs the query role <-?- principal\n";
    exit 1;
}
